No cursor in anvil

In anvil, the cursor is not visible. (The mouse works, since I can mark text using the mouse.) 

I get the following warning on the first startup of anvil (at subsequent starts of anvil I do not get this error):
`anvil::cursor: Unable to load xcursor: Theme has no default cursor, using fallback cursor`

Thanks for all that testing, I believe to have finally found the reason why this is happening.

Seems our test for `gbm_bo_create_with_modifiers2` fails, this is definitely something we need to look into as `mesa-23.1.9` should have it.

```
warning: smithay@0.3.0: In file included from test_gbm_bo_create_with_modifiers2.c:1:
warning: smithay@0.3.0: In file included from /nix/store/n0d2gbxcslrgdkiba49y3swvsj78jwfh-mesa-23.1.9-dev/include/gbm.h:34:
warning: smithay@0.3.0: In file included from /nix/store/nn2ypjvdr6frc1zrkf2w9hy45kp6cdb6-clang-wrapper-16.0.6/resource-root/include/stdint.h:52:
warning: smithay@0.3.0: In file included from /nix/store/jhi4wsbrxfscrf57k46d1lfq1v8d25kx-glibc-2.38-27-dev/include/stdint.h:26:
warning: smithay@0.3.0: In file included from /nix/store/jhi4wsbrxfscrf57k46d1lfq1v8d25kx-glibc-2.38-27-dev/include/bits/libc-header-start.h:33:
warning: smithay@0.3.0: /nix/store/jhi4wsbrxfscrf57k46d1lfq1v8d25kx-glibc-2.38-27-dev/include/features.h:414:4: error: _FORTIFY_SOURCE requires compiling with optimization (-O) [-Werror,-W#warnings]
warning: smithay@0.3.0: #  warning _FORTIFY_SOURCE requires compiling with optimization (-O)
warning: smithay@0.3.0:    ^
warning: smithay@0.3.0: 1 error generated.
```

With `gbm_bo_create_with_modifiers2` disabled we fall back to use `gbm_bo_create_with_modifiers` in our
[gbm allocator](https://github.com/Smithay/smithay/blob/fbe8ef20613d6417fc1c738bc32799aeee640d4e/src/backend/allocator/gbm.rs#L74). Unfortunately we do not check if any `flags` have been requested and silently drop then as `gbm_bo_create_with_modifiers` does not take flags. This is problematic for the cursor plane, or more specifically `gbm_bo_write`, as this is only guaranteed to work when the `GBM_BO_USE_WRITE` `flag` is passed to gbm.
When `GBM_BO_USE_WRITE` is passed to mesa gbm it will actually create a `dumb` buffer, which implicitly has a
mmap set and which is used in `gbm_bo_write`. 

That leaves the question why `gbm_bo_write` does not fail. The answer is, it does, but we fail to recognize it :) 
[cursor_buffer.write](https://github.com/Smithay/smithay/blob/fbe8ef20613d6417fc1c738bc32799aeee640d4e/src/backend/drm/compositor/mod.rs#L3079) has a nested result and we only check the outer result which only
tells us if the gbm device has been destroyed.

So we actually have three issues here:
1. the build script for detecting support for `gbm_bo_create_with_modifiers2` has issues (at least in this case here)
2. our fallback silently drops the flags
3. the error check for `gbm_bo_write` is wrong

2 and 3 should be quite easy to fix, 1 will probably be more work and.

cc @Drakulix

--- a/src/backend/drm/compositor/mod.rs
+++ b/src/backend/drm/compositor/mod.rs
@@ -3076,7 +3076,12 @@ where
             }
         };
 
-        if let Err(err) = cursor_buffer.write(data) {
+        let Ok(res) = cursor_buffer.write(data) else {
+            info!("failed to write cursor buffer, device destroyed");
+            return None;
+        };
+
+        if let Err(err) = res {
             info!("failed to write cursor buffer; {}", err);
             return None;
         }

